docs/testing-with-phpunit.md

```diff
@@ -0,0 +1,225 @@
+- `.tests/inc/class-test-sample.php` - A basic test to get you started.
+
+If you already have PHPUnit set up for a different environment you can temporarily move your existing `phpunit.xml.dist` file and a `.tests` directory to compare the scaffolded versions to your existing ones. The dot prefix ensures this directory is not accessible to HTTP requests.
+
+The scaffolded files can be edited as you need.
+
+
+## Running Tests
+
+With the above scaffolding in place running your unit tests is straightforward:
+
+```sh
+composer dev-tools phpunit
+```
+
+By default this will attempt to run your tests on the [Local Server](docs://local-server) environment. If you are using [Local Chassis](docs://local-chassis) add the `--chassis` flag eg:
+
+```sh
+composer dev-tools phpunit --chassis
+```
+
+### Passing Arguments To PHPUnit
+
+Often you may have multiple test directories, different configurations to run or you want to generate coverage reports. To pass any of the supported command line options to PHPUnit you need to add them after the options delimiter `--`. For example:
+
+```sh
+composer dev-tools phpunit -- --coverage-xml coverage --log-junit junit.xml
+```
+
+The [full list of PHPUnit command line options is available here](https://phpunit.readthedocs.io/en/7.1/textui.html) or you can run `composer dev-tools phpunit -- --help`.
+
+
+## Writing Tests
+
+On top of the standard PHPUnit framework Altis bundles the WP PHPUnit framework to provide advanced functionality for integration testing.
+
+### Naming And Organisation
+
+All test classes should be prefixed with `class-test-` and have a `.php` suffix. The classes themselves should be namespaced and be named to match the file name, for example the file `class-test-feature-a.php` should declare a class called `Test_Feature_A`.
+
+Within the `.tests` directory you can organise your test classes as you see fit however it is good practice to group related tests into directories. This makes it easier to run a subset of tests rather than the full suite.
+
+Group tests into a class when they test different aspects of the same piece of functionality. It’s especially convenient to put tests together in a class when they can share a common `setUp()` or `setUpBeforeClass()` routine. As a rule, a single test class should not contain tests for more than one function/method and should test every possible input and expected output.
+
+### Unit Tests
+
+To create a unit test your class must extend the `PHPUnit\FrameWork\TestCase` class and the methods that contain [assertions](https://phpunit.readthedocs.io/en/7.1/assertions.html) must start with `test`.
+
+```php
+<?php
+
+namespace Project\Tests;
+
+use PHPUnit\Framework\TestCase;
+
+class Test_Units extends TestCase {
+
+	/**
+	 * This special method is run once before all the tests in the class.
+	 */
+	public function setUpBeforeClass() {
+		require_once 'path/to/file/with/functions.php';
```

Comment:
Files should not be required inside of test classes. This is part of the environment setup, and should happen in `bootstrap.php`.